File: Anush_Qalashyan/Homework/11.11.2015/battle.cpp
#include <iostream>

int main ()
{
	int N = 10;
	int a[10][10]; 
	for (int i = 0; i < N; ++i) 
	{
		for (int j = 0; j < N; ++j)
		{
			a[i][j]=0;
		}
	}
	a[1][1]=1;
	a[1][2]=1;
	a[1][3]=1;
	a[2][4]=1;
	a[3][5]=1;
	a[4][2]=1;
	a[4][3]=1;
	a[5][2]=1;
	a[5][3]=1;

	int count = 0; 
	for (int i = 0; i < N; ++i)
	{
		for (int j = 0; j < N; ++j)
		{
			if (a[i][j] == 1) 
			{
				if (i==0) 
				{
					if (j==0)  
					{
						++count;
					}
					if ((j!= 0) &&  ( a[i][j-1] ==0)) 
					{
						++count; 
					}
				}  
				else 
				{
					if ((j == 0) && (a[i-1][j] == 0))
					{
						++count; 
					}
					if ((j != 0) && (a[i-1][j] == 0) && (a[i][j-1] == 0))
					{
						++count;
					}
				}
			}
		}
	}


	for (int i = 0; i < N; ++i)
	{
		for (int j = 0; j < N; ++j)
		{
			std::cout << a[i][j] << " ";
		}
		std::cout << std::endl;
	}
	std::cout << count << std::endl;

	return 0;
}

File: Edgar_Hakobyan/Homework/java/battle.java
import java.util.*;

class battle {
	public static int n;
	public static int width = 0;
	public static int length = 0;
	public static int boat = 0;
	public static int count = 0;
	public static void main(String[] args) {
	
		Scanner in = new Scanner(System.in);
		System.out.println("enter size of table");
		n = in.nextInt();
		int[][] array = new int[n][n];
		for(int i=0; i<n; ++i) {
			for(int j=0; j<n; ++j) {
				System.out.print("array["+i+"]["+j+"]=");
				array[i][j] = in.nextInt();
			}
		}
		for(int i=0; i<n; ++i) {
            for(int j=0; j<n; ++j) {
                System.out.print(array[i][j]);
            }
        	System.out.println();
		}
	
		for(int i=0; i<n; ++i) {
			for(int j=0; j<n; ++j) {
				if(array[i][j] == 1) {
					width = 1;
					length = 1;
					boat = 1;
					boat_verify(array,i,j);
					if(width * length == boat) {
						++count;
					}
				}
			}
		}
		System.out.println("count of boats = " + count);
	}
	
	public static void boat_verify(int[][] array,int i,int j) {
		array[i][j] = 2;
		if( (j < n-1) && (array[i][j+1] == 1) ) {
			++boat;
			++length;
			boat_verify(array,i,j+1);
		}
		if( (i < n-1) && (array[i+1][j] == 1) ) {
			++boat;
			++width;
			boat_verify(array,i+1,j);
		}
		if( (j > 0) && (array[i][j-1] == 1) ) {
			++boat;
			if(array[i-1][j-1] != 2) {
				++length;
			}
			boat_verify(array,i,j-1);
		}
		if( (i > 0) && (array[i-1][j] == 1) ) {
			++boat;
			if(array[i-1][j+1] != 2) {
				++width;
			}
			boat_verify(array,i-1,j);
		}
	}
}
